server/lib/system/system.getNetworkMode.js

```diff
@@ -14,9 +15,17 @@ async function getNetworkMode() {
   }
 
   if (!this.networkMode) {
-    const cmdResult = await exec('head -1 /proc/self/cgroup | cut -d/ -f3');
-    const [containerId] = cmdResult.split('\n');
-    const gladysContainer = this.dockerode.getContainer(containerId);
+    fs.access(`/var/lib/gladysassistant/containerId`, fs.constants.F_OK, (err) => {
+      if (err) {
+        // eslint-disable-next-line no-useless-escape
+        this.containerId = exec(
+          'cat /proc/self/cgroup | grep -o -e "docker.*" | head -n 1 | sed -e "s/.scope//g;s/docker-//g;s!docker/!!g"',
+        );
+      } else {
+        this.containerId = fs.readFile('/var/lib/gladysassistant/containerId', 'utf8');
```

Comment:
This won't work, you are using a callback (async) function, the result won't be there.

You need to use async/await.

Two possibilities:

- Promisify fs (for example with bluebird), because fs doesn't return promise by default.
- Use fs-extra, a lib we already use in Gladys: https://github.com/jprichardson/node-fs-extra

In Gladys 4 we don't use the callback pattern (because of the famous callback hell issue), and we rely only on async/await.

Reply:
OK understood

Reply:
@Pierre-Gilles  what about `const { promises: fs } = require("fs");` ?

Reply:
Indeed! I didn't know about this one, I learned something :D 

You'll need to change the access too btw